pdr_backend/lake/base_data_store.py

```diff
@@ -0,0 +1,22 @@
+from typing import Dict
+
+from enforce_typing import enforce_types
+
+
+class BaseDataStore:
+    @enforce_types
+    def __new__(cls, base_path: str):
+        if not hasattr(cls, "_instances"):
```

Comment:
this looks like some kind of singleton, but I think it needs more context/comments. And spacing between blocks and before the return statement.